File: 2026/07-July/07-Tuesday/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var sumAndMultiply = function (n) {
    const digits = [];

    while (n > 0) {
        const d = n % 10;
        if (d !== 0)
            digits.push(d);
        n = Math.floor(n / 10);
    }

    digits.reverse();

    let value = 0;
    let sum = 0;

    for (const d of digits) {
        value = value * 10 + d;
        sum += d;
    }

    return value * sum;
};

File: 2026/07-July/07-Tuesday/solution.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        digits = []

        while n > 0:
            d = n % 10
            if d:
                digits.append(d)
            n //= 10

        digits.reverse()

        value = 0
        digit_sum = 0

        for d in digits:
            value = value * 10 + d
            digit_sum += d

        return value * digit_sum
